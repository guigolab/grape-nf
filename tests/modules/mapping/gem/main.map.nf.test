nextflow_process {

    name "Test Process map"
    script "modules/mapping/gem/main.nf"
    process "map"

    tag "module"

    setup {
        run("index") {
            script "modules/mapping/gem/main.nf"
            process {
                """
                input[0] = file("${baseDir}/data/genome.fa")
                input[1] = file("${baseDir}/data/annotation.gtf")
                """
            }
        }
    }

    test("Should align reads to the reference using GEM (compressed annotation)") {

        when {
            params {
                maxMismatches = 4
                maxMultimaps = 10
            }
            process {
                """
                def fastqs = file("${baseDir}/data/test1_*.fastq.gz")
                def qScore = fastq(fastqs[0]).qualityScore()
                input[0] = file("${baseDir}/data/annotation.gtf.gz")
                input[1] = index.out
                input[2] = Channel.from(
                    [
                        [ "sample1", "test1", fastqs, "fastq", ["fqRd1", "fqRd2"], qScore]
                    ]
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should align reads to the reference using GEM (uncompressed annotation)") {

        when {
            params {
                maxMismatches = 4
                maxMultimaps = 10
            }
            process {
                """
                def fastqs = file("${baseDir}/data/test1_*.fastq.gz")
                def qScore = fastq(fastqs[0]).qualityScore()
                input[0] = file("${baseDir}/data/annotation.gtf")
                input[1] = index.out
                input[2] = Channel.from(
                    [
                        [ "sample1", "test1", fastqs, "fastq", ["fqRd1", "fqRd2"], qScore]
                    ]
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
