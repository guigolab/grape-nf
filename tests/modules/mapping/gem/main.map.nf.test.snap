{
    "Should align reads to the reference using GEM (compressed annotation)": {
        "content": [
            {
                "0": [
                    [
                        "sample1",
                        "test1",
                        "sample1_m4_n10_toGenome.bam:md5,48faeddd5179ce7b5d629ad937851b45",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    [
                        "sample1",
                        "test1",
                        "sample1_m4_n10_toGenome.bam.bai:md5,f0b712d99d68ae3132bdb287ed5a5e5d",
                        "bai",
                        "GenomeAlignmentsIndex",
                        true
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "genomeAlignments": [
                    [
                        "sample1",
                        "test1",
                        "sample1_m4_n10_toGenome.bam:md5,48faeddd5179ce7b5d629ad937851b45",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "genomeAlignmentsIndices": [
                    [
                        "sample1",
                        "test1",
                        "sample1_m4_n10_toGenome.bam.bai:md5,f0b712d99d68ae3132bdb287ed5a5e5d",
                        "bai",
                        "GenomeAlignmentsIndex",
                        true
                    ]
                ],
                "junctions": [
                    
                ],
                "stats": [
                    
                ],
                "transcriptomeAlignments": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-02T08:54:34.740923082"
    },
    "Should align reads to the reference using GEM (uncompressed annotation)": {
        "content": [
            {
                "0": [
                    [
                        "sample1",
                        "test1",
                        "sample1_m4_n10_toGenome.bam:md5,48faeddd5179ce7b5d629ad937851b45",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    [
                        "sample1",
                        "test1",
                        "sample1_m4_n10_toGenome.bam.bai:md5,f0b712d99d68ae3132bdb287ed5a5e5d",
                        "bai",
                        "GenomeAlignmentsIndex",
                        true
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "genomeAlignments": [
                    [
                        "sample1",
                        "test1",
                        "sample1_m4_n10_toGenome.bam:md5,48faeddd5179ce7b5d629ad937851b45",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "genomeAlignmentsIndices": [
                    [
                        "sample1",
                        "test1",
                        "sample1_m4_n10_toGenome.bam.bai:md5,f0b712d99d68ae3132bdb287ed5a5e5d",
                        "bai",
                        "GenomeAlignmentsIndex",
                        true
                    ]
                ],
                "junctions": [
                    
                ],
                "stats": [
                    
                ],
                "transcriptomeAlignments": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-02T08:55:30.059497532"
    }
}